credoai/modules/dataset_modules/dataset_base.py

```diff
@@ -156,6 +159,44 @@ def _calculate_mutual_information(self, categorical_features=None, normalize=Tru
 
         return results
     
+    def _assess_balance_metrics(self):
+        """Calculate dataset balance statistics and metrics 
 
-    
+        Returns
+        -------
+        dict
+            'sample_balance': distribution of samples across groups
+            'label_balance': distribution of labels across groups
+            'metrics': maximum statistical parity and maximum disparate impact between groups for all preferred label value possibilities 
+        """
+        credo_data = cl.CredoData(name='dataset',
```

Comment:
Oh, this is weird. CredoData should not be made within a module. I would redefine `_concat_features_label_to_dataframe` as [static method](https://www.geeksforgeeks.org/class-method-vs-static-method-python/). You can then use it here, as well as within CredoData in the future if we need.